codec-http/src/main/java/io/netty/handler/codec/http/CombinedHttpHeaders.java

```diff
@@ -64,6 +67,42 @@ public CharSequence escape(CharSequence value) {
     }
 
     @Override
+    public HttpHeaders add(Headers<? extends CharSequence, ? extends CharSequence, ?> headers) {
+        // Override the fast-copy mechanism used by DefaultHeaders
+        checkNotNull(headers, "headers");
+        if (headers == this) {
```

Comment:
consider moving the previous null check into `else` block of the `instanceof` check.